From 5e9fefe317942a5189e8f250e79e0b8dcfa7673b Mon Sep 17 00:00:00 2001
From: Kai-Heng Feng <kai.heng.feng@canonical.com>
Date: Fri, 8 Mar 2019 13:11:17 +0800
Subject: [PATCH 2/2] HID: Increase maximum report size allowed by
 hid_field_extract()

Commit 71f6fa90a353 ("HID: increase maximum global item tag report size
to 256") increases the max report size from 128 to 256.

We also need to update the report size in hid_field_extract() otherwise
it complains and truncates now valid report size:
[ 406.165461] hid-sensor-hub 001F:8086:22D8.0002: hid_field_extract() called with n (192) > 32! (kworker/5:1)

BugLink: https://bugs.launchpad.net/bugs/1818547
Fixes: 71f6fa90a353 ("HID: increase maximum global item tag report size to 256")
Signed-off-by: Kai-Heng Feng <kai.heng.feng@canonical.com>
Signed-off-by: Jiri Kosina <jkosina@suse.cz>

(cherry picked from commit 94a9992f7dbdfb28976b565af220e0c4a117144a)
[Reason for cherry-picking this commit:
  This is a fix commit of commit 71f6fa90a353 ("HID: increase maximum
  global item tag report size to 256") and I backported the commit
  before. So, also backport this fix commit.]
[Resolved conflict due to chromiumos' commit b6a7d777fcb9 ("UPSTREAM:
HID: core: fix dmesg flooding if report field larger than 32bit").]
Signed-off-by: Tsuchiya Yuto (kitakar5525) <kitakar@gmail.com>
---
 drivers/hid/hid-core.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/hid/hid-core.c b/drivers/hid/hid-core.c
index 3f6534cb97143..2e1107fd8df52 100644
--- a/drivers/hid/hid-core.c
+++ b/drivers/hid/hid-core.c
@@ -1374,10 +1374,10 @@ static u32 __extract(u8 *report, unsigned offset, int n)
 u32 hid_field_extract(const struct hid_device *hid, u8 *report,
 			unsigned offset, unsigned n)
 {
-	if (n > 32) {
-		hid_warn_once(hid, "%s() called with n (%d) > 32! (%s)\n",
+	if (n > 256) {
+		hid_warn_once(hid, "%s() called with n (%d) > 256! (%s)\n",
 			      __func__, n, current->comm);
-		n = 32;
+		n = 256;
 	}
 
 	return __extract(report, offset, n);
-- 
2.27.0

